src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -502,6 +502,13 @@ private void ProcessProjectChange(Solution solution, ProjectId projectId)
                 // Nuget may throw an ArgumentException when there is something about the project 
                 // they do not like/support.
             }
+            catch (InvalidOperationException)
```

Comment:
Just curious, have you considered the other approach @davkean suggested? (https://github.com/dotnet/roslyn/issues/20101#issuecomment-441157188)

Reply:
I'm fine with ignoring the exception, just wanna make sure you considered the alternative as well. For project related questions, you might want to ask @jasonmalinowski and @davkean. Here's a link about design-time build
https://github.com/dotnet/project-system/blob/master/docs/design-time-builds.md

Reply:
Got it, is it enough to call [TryGetCompilation](http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Workspace/Solution/Project.cs,8683c8e64484d05d) ? If so, I don't see why that can't be done as well. Once compilation is available, do we get a project or solution change notification?

Reply:
You can get a compilation without a design time build having completed, you just won't get the full information. @davkean how would you recommend we know it's complete? Even if you've completed on your end you might not have pushed it to us yet. The only thing I can think of is to check the bit you pass for IWorkspaceProjectContext.LastDesignTimeBuildSucceeded, but that doesn't really distinguish between "one happened" and "one succeeded".

Reply:
Maybe just use a heuristic "if the project has no metadata references, it then must not have any nuget packages (or at least none that we care about)?"

Reply:
we don't have API for this yet. platform team is working on an API for this kind of stuff. not specific to design time build but, more like, is solution/project actually fully loaded API which include project system, nuget, msbuild, vs, roslyn and etc.

Reply:
@ryzngard @heejaechang Any reason not to use the "are there no binaries" as a heuristic until that point? Since we're not able to fix this from this PR and we're already well in the realm of "tactical fixes" should we just do that?